### ISSUE TITLE
Ordering   does  not work at version 0.9.0-beta2

### ISSUE THREAD

**Issue by Lumax-Rus** [pivotal-energy-solutions/django-datatable-view#113]
```
class Task(models.Model):
    CANCELLED = -1
    REQUIRES_ATTENTION = 0
    WORK_IN_PROGRESS = 1
    COMPLETE = 2

    class Meta:
        ordering = ['-id']

    Status = (
        (CANCELLED, 'Cancelled'),
        (REQUIRES_ATTENTION, 'Requires attention'),
        (WORK_IN_PROGRESS, 'Work in progress'),
        (COMPLETE, 'Complete'),
    )
    id = models.AutoField(primary_key=True)
    name = models.TextField()
    status = models.IntegerField(choices=Status, default=REQUIRES_ATTENTION)
```

This is work at version 0.8.3, but with bug. At the table field id sorting only once. 
I fixed this bug by editing  the file `datatableview.js` 
My `datatableview.js`:

```
var datatableview = {
    auto_initialize: true,
    defaults: {
        "bServerSide": true
    },

    getCookie: function(name) {
        var cookieValue = null;
        if (document.cookie && document.cookie != '') {
            var cookies = document.cookie.split(';');
            for (var i = 0; i < cookies.length; i++) {
                var cookie = jQuery.trim(cookies[i]);
                // Does this cookie string begin with the name we want?
                if (cookie.substring(0, name.length + 1) == (name + '=')) {
                    cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
                    break;
                }
            }
        }
        return cookieValue;
    },

    initialize: function($$, opts) {
        if (typeof window.console === "undefined" || typeof window.console.log === "undefined") {
            console = {
                log: function(){},
                info: function(){}
            };
        }
        var options_name_map = {
            'sortable': 'bSortable',
            'sorting': 'aaSorting',
            'visible': 'bVisible'
        };

        var template_clear_button = $('<a href="#" class="clear-search">Clear</a>');

        var initialized_datatables = [];
        $$.each(function(){
            var datatable = $(this);
            var column_options = [];
            var sorting_options = [];

            datatable.find('thead th').each(function(){
                var header = $(this);
                var options = {};
                for (var i = 0; i < header[0].attributes.length; i++) {
                    var attr = header[0].attributes[i];
                    if (attr.specified && /^data-/.test(attr.name)) {
                        var name = attr.name.replace(/^data-/, '');
                        var value = attr.value;

                        // Typecasting out of string
                        name = options_name_map[name];
                        if (/^b/.test(name)) {
                            value = (value === 'true');
                            options[name] = value;
                        }

                        if (name == 'aaSorting') {
                            // This doesn't go in the column_options
                            var sort_info = value.split(',');
                            sort_info[1] = parseInt(sort_info[1]);
                            sorting_options.push(sort_info);
                            continue;
                        }


                    }
                }
                options["orderSequence"]=[ "desc", "asc" ];
                column_options.push(options);

            });

            //Arrange the sorting column requests and strip the priority information
            sorting_options.sort(function(a, b){ return a[0] - b[0] });
            for (var i = 0; i < sorting_options.length; i++) {
                sorting_options[i] = sorting_options[i].slice(1);
            }

            options = $.extend({}, datatableview.defaults, opts, {
                "order": sorting_options,
                "columns": column_options,
                "orderFixed": sorting_options,
                "sAjaxSource": datatable.attr('data-source-url')
            });
            try {
                options = confirm_datatable_options(options, datatable);
            } catch (e) {

            }

            var initialized_datatable = datatable.dataTable(options);
            initialized_datatables.push(initialized_datatable[0]);

            try {
                initialized_datatable.fnSetFilteringDelay();
            } catch (e) {
                console.info("datatable plugin fnSetFilteringDelay not available");
            }

            var search_input = initialized_datatable.closest('.dataTables_wrapper').find('.dataTables_filter input');
            var clear_button = template_clear_button.clone().click(function(){
                $(this).trigger('clear.datatable', [initialized_datatable]);
                return false;
            }).bind('clear.datatable', function(){
                search_input.val('').keyup();
            });
            search_input.after(clear_button).after(' ');
        });
        return $(initialized_datatables).dataTable();
    }
}

$(function(){
    if (datatableview.auto_initialize) {
        datatableview.initialize($('.datatable'));
    }
});
```

**Comment by tiliv** [comment]
Are you by chance using datatables.js 1.10.9?  Somewhere in the updates they started using a jQuery ``.data('sorting', xxx)`` storage on the ``<th>``, and it gets shadowed by our ``data-sorting`` html attribute.  The result is that the table loads okay, but if you try to reverse the sort of the default ordering column, it sort of freaks out.

If this is the case, if you inspect your XHR requests, you should see that each click on the header sends a different value for ``sSortDir_0`` (or instead of ``0``, whichever column it is), probably rotating through values like ``a``, ``s``, ``c``, ``,``, etc.  It's incorrectly looping over the text we're storing on that attribute.

If this really is the issue you discovered, it is solved in 0.9, but if you'd like to confirm, that would be great.

**Comment by tiliv** [comment]
I forgot to mention, I wasn't sure which version of ``django-datatable-view`` you're using.  The title says the 0.9 beta, but in the text you mentioned only 0.8.

Is there a different bug in 0.9 that you wanted to report?
